### DISCUSSION TITLE
Wrong background color in delta pager inside lazygit

### DISCUSSION THREAD

**Discussion by ComplexPlane** [ghostty-org/ghostty#3962] [Feature Requests, Ideas] [answered]
I'm asking this here because I'm not 100% sure who to blame and I figure Ghostty devs' perspective could be valuable.

[lazygit](https://github.com/jesseduffield/lazygit) allows you to use an alternative pager for viewing diffs, including [delta](https://github.com/dandavison/delta). It seems that when this is done in Ghostty, the line-wise background color for additions is gray instead of green:

<img width="1087" alt="image" src="https://github.com/user-attachments/assets/a28b9671-4d7a-4d8f-9ba8-69d4bffc4e13" />

It should look like:

<img width="1087" alt="image" src="https://github.com/user-attachments/assets/ee6ed82d-5f0d-4872-a487-135c7133e6a5" />

I wonder if this is related? https://github.com/jesseduffield/lazygit/issues/2962

## Reproduction Notes

Lazygit config:

```yaml
git:
  paging:
    colorArg: always
    pager: delta --dark --paging=never --line-numbers --hyperlinks --hyperlinks-file-link-format="lazygit-edit://{path}:{line}"
```

Here's a summary of the background color values I observe in different situations, including running delta outside lazygit, overriding TERM, and in kitty:

| Terminal | Command                  | Background Color       |
| -------- | ------------------------ | ---------------------- |
| ghostty  | lg                       | Palette 235 (`#262626`)  |
| ghostty  | TERM=xterm-kitty lg      | `#002800`                |
| ghostty  | delta                    | `#002800`                |
| ghostty  | delta --true-color never | Palette 22 (`#005f00`)   |
| kitty    | lg                       | `#002800` (color picker) |

**Comment by 00-kat** [comment]
Do make sure that you have [terminfo] properly installed, just to be sure.

[terminfo]: https://ghostty.org/docs/help/terminfo

**Reply by ComplexPlane** [reply]
Seems like it, $TERMINFO points to what seems like the right path and $TERM is indeed xterm-ghostty

**Comment by chris4prez** [comment]
I'm running into this too, and def think it has to do with this: https://github.com/jesseduffield/lazygit/issues/2962 vs being a ghostty issue.

I've also seen similar weirdness with how lazygit handles diff colors in tmux in other terminals too.

**Reply by 00-kat** [reply]
Seems like that issue was closed, is this still reproducible?

**Reply by ComplexPlane** [reply, marked as answer]
It appears fixed on my end:

```
lg --version
commit=c03b89227092b852d50015d289a7c6d8c69811c5, build date=2025-01-17T13:37:10Z, build source=binaryRelease, version=0.45.2, os=darwin, arch=arm64, git version=2.39.5 (Apple Git-154)
```

```
ghostty --version
Ghostty 1.1.0

Version
  - version: 1.1.0
  - channel: stable
Build Config
  - Zig version: 0.13.0
  - build mode : builtin.OptimizeMode.ReleaseFast
  - app runtime: apprt.Runtime.none
  - font engine: font.main.Backend.coretext
  - renderer   : renderer.Metal
  - libxev     : main.Backend.kqueue
```
